server/test/Comment.test.js

```diff
@@ -0,0 +1,124 @@
+import mongoose from 'mongoose';
+import dotenv from 'dotenv';
+import app from '../app.js';
+import request from "supertest";
+import User from '../models/userModel.js';
+import Post from '../models/postModel.js';
+import Comment from '../models/commentModel.js';
+
+
+dotenv.config();
+
+describe('/api/comments', () => {
+
+    let user;
+    let post;
+    let comment;
+    beforeAll(async () => {
+        await mongoose.connect(process.env.MONGODB_URI_TEST, {
+            useNewUrlParser: true,
+            useUnifiedTopology: true,
+        });
+
+        user = await User.create(
+            { username: 'user', email: 'user@user.com', password: '123User!' }
+        );
+
+        await User.create(
+            { username: 'secondUser', email: 'user2@user.com', password: '123User!' }
+        );
+
+        post = await Post.create({
+            userId: user._id,
+            profileName: user.username,
+            postContent: 'placeholder'
+        });
+
+        comment = await Comment.create({
+            postId: post._id,
+            userId: user._id,
+            profileName: user.username,
+            commentContent: 'placeholder'
+        });
+    });
+
+    afterAll(async () => {
+        await mongoose.disconnect();
+    });
+
+    describe('Create a comment in db', () => {
+        it('should response with status code 201', async () => {
+            const user = await request(app).post('/api/users/auth').send({ email: 'user@user.com', password: '123User!' });
+
+            const response = await request(app).post(`/api/comments?postId=${post._id}`).send({ message: 'test' }).set('Cookie', user.header['set-cookie']);
```

Comment:
Correct me if I'm wrong - you're testing if the api call works but... are you testing the functions you created ?
You're making api calls from scratch and not testing the functions ?

for example the function `postResponse` in `CommentForm` , Are you testing the FUNCTION here ? because i dont see it anywhere here.

it's like youre building the functions here instead of testing the already existing functions in your code ?

Reply:
i dont quick get what you meant, but im using app.js which has all the code and posting to the api path and checking for its response to match what i expect.

Reply:
I see, no problem! Let's try this way:
In the whole tests that you coded, are you testing all your server-side functions ? Because I didn't see you use a single one of them. The functions I am talking about:
Post : `createPost` , `getPost` , `deletePost`
Comment: `createComment` , `getComment`, `deleteComment`
User : `authUser` , `registerUser`, `logoutUser` , `getUser`
Vote: `votePost` , `getVotes`
?

The main focus of testing, is to test these functions if they work and return exactly as they are designed ?

Reply:
When I request app it goes through all the code and routes to the corresponding function that's why if you look for example into voteRoutes.js you see I have the middleware protect which lets it access to authorized users only which is why I set a cookie from the user that I logged in and after that it goes to for this example to `createComment`